go-libp2p-blossomsub/pubsub.go

```diff
@@ -26,8 +26,8 @@ import (
 	logging "github.com/ipfs/go-log/v2"
 )
 
-// DefaultSoftMaxMessageSize is 10 MiB.
-const DefaultSoftMaxMessageSize = 10 << 20
+// DefaultSoftMaxMessageSize is 1 MiB.
+const DefaultSoftMaxMessageSize = 1 << 20
 
 // DefaultHardMaxMessageSize is 20 MB.
 const DefaultHardMaxMessageSize = 10 << 21
```

Comment:
Increasing it is not panacea. The bigger we make it, the easier it is for potential attackers to create broadcast storms of bad messages with large sizes and to consume excessive bandwidth. All of the scoring in Gossip/BlossomSub are based on message counts / rates, and not bandwidth effectively spent on a message, under the assumption that large messages are not really what is broadcasted via PubSub (and instead the PubSub is used only to broadcast that the peer _has_ the data, not the data itself).

It also doesn't solve our large frame issue - middle peers which are not updated will still drop these jumbo frames, and it takes quite some time until peers are updated, and by that time, we can do fragmentation.

I personally want this value back to something like 2 or 4 MiB - it is way too big to be useful, and we see this today with the 17-18MiB messages that take a minute to propagate.

Reply:
wouldn't the problem persist with fragmentation, given we drop non-conformant messages?

Reply:
That's correct, and it is why I've used a different bitmask (4 leading zeroes) for the fragment bitmask.

There is no real backwards compatible option here:
- Frames bigger than 20 MiB can be synced only from the beacon, because the beacon has increased limits. `2.0.4-p2` has a 20 MiB limit (what this PR is about).
- Even if the limit is increased on the beacon, and the middle nodes are also updated, if the receiver is not updated, they cannot receive the frame whose size exceeds 20MiB.

Even nodes which are altruistic enough today to set their rate limits to more than 10 unique peers per minute cannot send frames above 20MiB, due to the gRPC message size limits.